Docs/details/events.md

```diff
@@ -0,0 +1,153 @@
+# Events
+
+Events are a defined set of data, core to how Santa interacts with a sync-server. Events are generated when there is a blocked `execve()` while in lockdown or monitor mode. Events are also generated in monitor mode for an `execve()` that was allowed to run, but would have been blocked in lockdown mode. This allows an admin to roll out Santa to their macOS fleet in monitor mode, but still collect meaningful data. The events collected while in monitor mode can be used to build a reasonably comprehensive whitelist of signing certificates and binaries before switching the fleet to lockdown mode.
+
+##### Event Data
+
+Events begin their life as an [SNTStoredEvent](https://github.com/google/santa/blob/master/Source/common/SNTStoredEvent.h) object. The SNTStoredEvent class is just a simple storage class that has properties for all the relevant bits of information. More importantly the class implements the [NSSecureCoding](https://developer.apple.com/documentation/foundation/nssecurecoding?language=objc) protocol. This allows the objects to be encoded and decoded for storage in the events sqlite3 database on disk and sent over XPC to another process.
+
+###### Archived Object
```

Comment:
I'd suggest removing this section entirely and just add a note to the end of the section above saying that the events are stored in a database temporarily until they can be uploaded and that the format is subject to change so don't use it. I'd maybe even suggest that if users have a need to get data out of the database without a sync server to raise an issue requesting that as a feature.